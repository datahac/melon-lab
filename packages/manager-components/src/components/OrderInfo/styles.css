.order-info {
  width: 100%;
  border: 1px solid var(--color-border);
}

.order-info__prices {
  border-bottom: 1px solid var(--color-border);
}

.order-info__prices,
.order-info__balances {
  display: flex;
  justify-content: space-between;
  padding: var(--space-s);
  align-items: center;
}

.order-info__price-desc {
  display: block;
}

.order-info__price-desc,
.order-info__balance-desc {
  font-size: var(--font-size-s);
  color: var(--color-grey);
}
