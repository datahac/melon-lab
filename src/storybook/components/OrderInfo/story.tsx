import { storiesOf } from '@storybook/react';
import React from 'react';
import OrderInfo from './index';

const data = {
  lastPrice: 0.5,
  bid: 0,
  ask: 0,
  tokens: {
    baseToken: {
      name: 'ETH-T',
      balance: 30.0,
    },
    quoteToken: {
      name: 'MLN-T',
      balance: 20.0,
    },
  },
};

storiesOf('Components|Order Info', module).add('Default', () => {
  return <OrderInfo {...data} />;
});
