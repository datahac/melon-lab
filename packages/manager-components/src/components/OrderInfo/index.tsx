import React, { StatelessComponent } from 'react';
import NumberFormat from 'react-number-format';

import styles from './styles.css';

export interface OrderInfoProps {
  ask?: number;
  bid?: number;
  lastPrice?: number;
  tokens: {
    [key: string]: {
      name: string;
      balance: number;
    };
  };
}

const OrderInfo: StatelessComponent<OrderInfoProps> = ({
  ask,
  bid,
  lastPrice,
  tokens,
}) => (
  <div className="order-info">
    <style jsx>{styles}</style>
    <div className="order-info__prices">
      <div className="order-info__last-price">
        {lastPrice ? (
          <NumberFormat
            value={lastPrice}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType="text"
            thousandSeparator={true}
          />
        ) : (
          <span>N/A</span>
        )}
        <span className="order-info__price-desc">Last Price</span>
      </div>
      <div className="order-info__bid">
        {bid ? (
          <NumberFormat
            value={bid}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType="text"
            type="text"
            thousandSeparator={true}
          />
        ) : (
          <span>N/A</span>
        )}
        <span className="order-info__price-desc">Bid</span>
      </div>
      <div className="order-info__ask">
        {ask ? (
          <NumberFormat
            value={ask}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType="text"
            thousandSeparator={true}
          />
        ) : (
          <span>N/A</span>
        )}
        <span className="order-info__price-desc">Ask</span>
      </div>
    </div>
    <div className="order-info__balances">
      <div className="order-info__balance">
        <span className="order-info__balance-desc">
          {tokens.baseToken.name}:
        </span>{' '}
        <NumberFormat
          value={tokens.baseToken.balance}
          decimalScale={4}
          fixedDecimalScale={true}
          displayType="text"
          thousandSeparator={true}
        />
      </div>
      <div className="order-info__balance">
        <span className="order-info__balance-desc">
          {tokens.quoteToken.name}:
        </span>{' '}
        <NumberFormat
          value={tokens.quoteToken.balance}
          decimalScale={4}
          fixedDecimalScale={true}
          displayType="text"
          thousandSeparator={true}
        />
      </div>
    </div>
  </div>
);

export default OrderInfo;
