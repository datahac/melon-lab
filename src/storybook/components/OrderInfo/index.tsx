import React, { StatelessComponent, Fragment } from 'react';
import * as Tm from '@melonproject/token-math';
import * as R from 'ramda';

import styles from './styles.css';

export interface OrderInfoProps {
  ask?: Tm.PriceInterface;
  bid?: Tm.PriceInterface;
  lastPrice?: Tm.PriceInterface;
  baseToken: Tm.QuantityInterface;
  quoteToken: Tm.QuantityInterface;
}

const OrderInfo: StatelessComponent<OrderInfoProps> = ({
  ask,
  bid,
  lastPrice,
  baseToken,
  quoteToken,
}) => (
  <div className="order-info">
    <style jsx>{styles}</style>
    <div className="order-info__prices">
      <div className="order-info__last-price">
        {lastPrice ? (
          <Fragment>{Tm.toFixed(lastPrice)}</Fragment>
        ) : (
          <span>N/A</span>
        )}
        <span className="order-info__price-desc">Last Price</span>
      </div>
      <div className="order-info__bid">
        {bid ? <Fragment>{Tm.toFixed(bid)}</Fragment> : <span>N/A</span>}
        <span className="order-info__price-desc">Bid</span>
      </div>
      <div className="order-info__ask">
        {ask ? <Fragment>{Tm.toFixed(ask)}</Fragment> : <span>N/A</span>}
        <span className="order-info__price-desc">Ask</span>
      </div>
    </div>
    <div className="order-info__balances">
      <div className="order-info__balance">
        <span className="order-info__balance-desc">
          {R.path(['token', 'symbol'], baseToken)}:
        </span>{' '}
        {baseToken && Tm.toFixed(baseToken)}
      </div>
      <div className="order-info__balance">
        <span className="order-info__balance-desc">
          {R.path(['token', 'symbol'], quoteToken)}:
        </span>{' '}
        {quoteToken && Tm.toFixed(quoteToken)}
      </div>
    </div>
  </div>
);

export default OrderInfo;
