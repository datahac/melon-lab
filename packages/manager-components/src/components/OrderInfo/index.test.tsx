import React from 'react';
import OrderInfo from './index';

const data = {
  lastPrice: 0.5,
  bid: 0,
  ask: 0,
  tokens: {
    baseToken: {
      name: 'ETH-T',
      balance: 30.0,
    },
    quoteToken: {
      name: 'MLN-T',
      balance: 20.0,
    },
  },
};

describe('Order Info', () => {
  const defaultElement = <OrderInfo {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
