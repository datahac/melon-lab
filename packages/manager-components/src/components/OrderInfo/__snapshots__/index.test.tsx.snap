// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Info should render correctly 1`] = `
<div
  className="jsx-__scopedHash order-info"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash order-info__prices"
  >
    <div
      className="jsx-__scopedHash order-info__last-price"
    >
      <NumberFormat
        allowEmptyFormatting={false}
        allowNegative={true}
        decimalScale={4}
        decimalSeparator="."
        displayType="text"
        fixedDecimalScale={true}
        getInputRef={[Function]}
        isAllowed={[Function]}
        isNumericString={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        onValueChange={[Function]}
        prefix=""
        suffix=""
        thousandSeparator={true}
        type="text"
        value={0.5}
      />
      <span
        className="jsx-__scopedHash order-info__price-desc"
      >
        Last Price
      </span>
    </div>
    <div
      className="jsx-__scopedHash order-info__bid"
    >
      <span
        className="jsx-__scopedHash"
      >
        N/A
      </span>
      <span
        className="jsx-__scopedHash order-info__price-desc"
      >
        Bid
      </span>
    </div>
    <div
      className="jsx-__scopedHash order-info__ask"
    >
      <span
        className="jsx-__scopedHash"
      >
        N/A
      </span>
      <span
        className="jsx-__scopedHash order-info__price-desc"
      >
        Ask
      </span>
    </div>
  </div>
  <div
    className="jsx-__scopedHash order-info__balances"
  >
    <div
      className="jsx-__scopedHash order-info__balance"
    >
      <span
        className="jsx-__scopedHash order-info__balance-desc"
      >
        ETH-T
        :
      </span>
       
      <NumberFormat
        allowEmptyFormatting={false}
        allowNegative={true}
        decimalScale={4}
        decimalSeparator="."
        displayType="text"
        fixedDecimalScale={true}
        getInputRef={[Function]}
        isAllowed={[Function]}
        isNumericString={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        onValueChange={[Function]}
        prefix=""
        suffix=""
        thousandSeparator={true}
        type="text"
        value={30}
      />
    </div>
    <div
      className="jsx-__scopedHash order-info__balance"
    >
      <span
        className="jsx-__scopedHash order-info__balance-desc"
      >
        MLN-T
        :
      </span>
       
      <NumberFormat
        allowEmptyFormatting={false}
        allowNegative={true}
        decimalScale={4}
        decimalSeparator="."
        displayType="text"
        fixedDecimalScale={true}
        getInputRef={[Function]}
        isAllowed={[Function]}
        isNumericString={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        onValueChange={[Function]}
        prefix=""
        suffix=""
        thousandSeparator={true}
        type="text"
        value={20}
      />
    </div>
  </div>
</div>
`;
